### PR TITLE
Use whole unsigned VC as VC proof signature payload

### PR DESCRIPTION
**Pull request by kziemianek**
* Changes the input for VC proof signature calculation from enclave's mrenclave to VC json.
* Removes all logic/calls from `VCManagement` related to `VCRegistry`. 

After this change newly issued VC content can be verified just by verifying proof's signature, there would be no need to reach out to onchain's `VCRegistry`.

### CODE REVIEW

File: tee-worker/ts-tests/integration-tests/common/utils/assertion.ts [hunk 1]
```diff
@@ -244,64 +244,71 @@ export async function assertIdGraphMutationResult(
...
     const { proof, ...vcWithoutProof } = vcPayloadJson;
-    assert.equal(vcIndex, vcPayloadJson.id, 'Check VcIndex error: VcIndex should be equal to vcPayload id');
 
     // step 5
+    // prepare teerex enclave registry data for further checks
     const enclaveCount = await context.api.query.teerex.enclaveCount();
+    const parachainBlockHash = await context.api.query.system.blockHash(vcPayloadJson.parachainBlockNumber);
+    const apiAtVcIssuedBlock = await context.api.at(parachainBlockHash);
+    const lastRegisteredEnclave = (await apiAtVcIssuedBlock.query.teerex.enclaveRegistry(enclaveCount))
+        .value as TeerexPrimitivesEnclave;
```
Comment (Kailai-Wang) [inline tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: I think checking `scheduledEnclave` should be more accurate - enclave registration needs to comply with it too. Right now I can't think of any case where `scheduledEnclave` and `enclaveRegistry` don't match though

Btw even if we want to check `enclaveRegistry` we should `enclaveCount` from the `parachainBlockHash` height

**Review by Kailai-Wang — COMMENTED** [review]
Nice and clean, not much to complain, thanks!

About the registry - this PR removes it completely. I'm thinking in the future, we might want to add "publish vc" feature where the user can link the selected VCs to their DID to make them public, so that anyone can see them by tracking down the DID. For that we'll need (again) a registry 🤔

File: tee-worker/ts-tests/integration-tests/common/utils/assertion.ts (hunk 1 above)
Comment (kziemianek) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: But Teerex's `ScheduledEnclave` is empty so there is no value to compare against :(

**Comment by kziemianek** [comment]
> Nice and clean, not much to complain, thanks!
> 
> About the registry - this PR removes it completely. I'm thinking in the future, we might want to add "publish vc" feature where the user can link the selected VCs to their DID to make them public, so that anyone can see them by tracking down the DID. For that we'll need (again) a registry 🤔

`VCRegistry` remains untouched - previously generated VCs can still be validated using it. It just can't be altered anymore because logic/calls are gone. I'm still waiting for decision on what we are going to do with old VCs.

File: tee-worker/ts-tests/integration-tests/common/utils/assertion.ts (hunk 1 above)
Comment (Kailai-Wang) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: Ah in `dev` it's not populated, but in production it shouldn't be empty

Comment (Kailai-Wang) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: Maybe we should add the scheduled enclave even in `dev` - like here: https://github.com/litentry/litentry-parachain/pull/2382/files#diff-f941434353cbdda40f528800d59a89b42e5e51396b03c092bf41ef8e294d8fd5

Comment (kziemianek) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: Maybe we can always populate it with worker's mrenclave for first validateer in dev see: https://github.com/litentry/litentry-parachain/pull/2382#issuecomment-1892237305

Comment (kziemianek) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: @Kailai-Wang, @felixfaisal  what do you think?

Comment (jonalvarezz) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: fyi: the vc-sdk uses `scheduledEnclave`'s value

File: tee-worker/ts-tests/integration-tests/common/utils/assertion.ts [hunk 2]
```diff
@@ -244,64 +244,72 @@ export async function assertIdGraphMutationResult(
...
+    // extrac proof and vc without proof json
     const vcPayloadJson = JSON.parse(decryptVcPayload);
     const { proof, ...vcWithoutProof } = vcPayloadJson;
-    assert.equal(vcIndex, vcPayloadJson.id, 'Check VcIndex error: VcIndex should be equal to vcPayload id');
 
     // step 5
-    const enclaveCount = await context.api.query.teerex.enclaveCount();
+    // prepare teerex enclave registry data for further checks
+    const parachainBlockHash = await context.api.query.system.blockHash(vcPayloadJson.parachainBlockNumber);
+    const apiAtVcIssuedBlock = await context.api.at(parachainBlockHash);
```
Comment (jonalvarezz) [inline tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:280]: `parachainBlockNumber` is the block used to calculate the VC, and it won't match the VC's Issuance block. Is it likely that the registered enclave changes between those two instances? better said, should we account for such a case?

**Review by jonalvarezz — APPROVED** [review]
lgtm

File: tee-worker/ts-tests/integration-tests/common/utils/assertion.ts (hunk 1 above)
Comment (BillyWooo) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:284]: @Kailai-Wang @kziemianek I think we can always populate it for first validateer.

**Comment by Kailai-Wang** [comment]
> > Nice and clean, not much to complain, thanks!
> > About the registry - this PR removes it completely. I'm thinking in the future, we might want to add "publish vc" feature where the user can link the selected VCs to their DID to make them public, so that anyone can see them by tracking down the DID. For that we'll need (again) a registry 🤔
> 
> `VCRegistry` remains untouched - previously generated VCs can still be validated using it. It just can't be altered anymore because logic/calls are gone. I'm still waiting for decision on what we are going to do with old VCs.

Yea I'm not talking about old VCs, but rather in the future - but OK we can always add it back whenever needed.

**Comment by kziemianek** [comment]
I will delete `VCRegistry` storage from `VCManagement` pallet.

**Review by Kailai-Wang — APPROVED** [review]

**Review by BillyWooo — APPROVED** [review]

**Comment by kziemianek** [comment]
Requested review again as there are new changes.

File: pallets/vc-management/src/schema.rs [hunk 3]
```diff
@@ -17,12 +17,21 @@
...
 
-use crate::{vc_context::Status, Config};
+use crate::Config;
 use codec::{Decode, Encode, MaxEncodedLen};
 use core_primitives::{SchemaContentString, SchemaIdString};
 use scale_info::TypeInfo;
 use sp_std::vec::Vec;
 
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub enum Status {
```
Comment (jonalvarezz) [inline pallets/vc-management/src/schema.rs:27]: just wondering why we need to keep this one

**Review by jonalvarezz — APPROVED** [review]
set it on fire 🔥

File: pallets/vc-management/src/schema.rs (hunk 3 above)
Comment (kziemianek) [reply pallets/vc-management/src/schema.rs:27]: It's used by `VCSchema` struct defined below.

File: tee-worker/ts-tests/integration-tests/common/utils/assertion.ts (hunk 2 above)
Comment (kziemianek) [reply tee-worker/ts-tests/integration-tests/common/utils/assertion.ts:280]: I think the time of VC's issuance is not important here - it doesn't matter what happens next after VC was generated as long as the VC was generated by legit enclave, but I would love to hear other opinions too.

File: pallets/vc-management/src/schema.rs (hunk 3 above)
Comment (jonalvarezz) [reply pallets/vc-management/src/schema.rs:27]: Right, but after removing the vc-registry, do we still need those statuses?

Comment (kziemianek) [reply pallets/vc-management/src/schema.rs:27]: There were no plans of removing schema/schema's status so this enum cannot be deleted. You think we should remove it ?

Comment (jonalvarezz) [reply pallets/vc-management/src/schema.rs:27]: Ah, I guess I'm missing something. If it is not part of the vc-registry then, sure, it is out of the scope 👍

**Comment by kziemianek** [comment]
I'll update weights and merge it.

**Pull request merged**